ignite/metrics/mean_average_precision.py

```diff
@@ -220,13 +220,16 @@ def __init__(
         .. versionadded:: 0.5.2
         """
 
-        super(MeanAveragePrecision, self).__init__(
+        # Initialize _BaseClassification first
```

Comment:
I'm not totally following why we should do this change. If you can please explain it, would be great! Thanks!
Can we do this in a separate PR?

Reply:
I think I got some issues with mypy of linting, but I think this is not needed indeed!